Edge/edge on kubernetes (#1399)
SoftwareAG/c8y-docs: content/edge-k8s/edge-k8s-troubleshooting-bundle/edge-k8s-cr-status-install-update-loop-backoff.md

---
weight: 10
title: Edge CR status InstallLoopBackOff or UpdateLoopBackOff
layout: redirect
---

> After deploying or updating {{< product-c8y-iot >}} Edge, if the Edge CR status appears as **InstallationLoopBackOff** or **UpdateLoopBackOff**, as shown below: 
> 
> Output of `kubectl get edge c8yedge -n c8yedge`:

Reviewer: ```suggestion
After deploying or updating {{< product-c8y-iot >}} Edge, if the Edge CR status is `InstallationLoopBackOff` or `UpdateLoopBackOff`, as part of the output of `kubectl get edge c8yedge -n c8yedge`:
```

```shell
NAME         DOMAIN NAME     VERSION     STATUS     
c8yedge      myown.iot.com   1017.0.0    InstallLoopBackOff
```

> Describe the Edge CR (cumulocity-iot-edge) `kubectl describe edge c8yedge -n c8yedge`.

Reviewer: ```suggestion
Describe the Edge CR (cumulocity-iot-edge) with the command `kubectl describe edge c8yedge -n c8yedge`.
```

In the events section, you will find the reasons for the install or update failure. For example: 

```yaml
Name: c8yedge
Namespace: c8yedge
API Version: edge.cumulocity.com/v1 
Kind: CumulocityIoTEdge

Status: 
	Deploying Generation: x
	State: InstallLoopBackOff 

Events: 
 Type    Reason               Age    From               Message 
 ------  ------               ----   ----               ------- 
 Normal  Validating           8m37s  cumulocityiotedge  validating 
 Normal  ValidationPassed     8m37s  cumulocityiotedge  validation passed 
 Normal  Installing           8m34s  cumulocityiotedge  installing 
…… 
……
 Normal  InstallLoopBackOff   1m00s  cumulocityiotedge  waiting for mongo server to be ready 
```

> Check the logs of the Edge Operator, MongoDB and Core pods to get more insights. 

Reviewer: ```suggestion
Check the logs of the Edge Operator, MongoDB and core pods to get more insights. 
```
